addons/mail/static/src/core/common/mail_core_common_service.js

```diff
@@ -29,7 +29,9 @@ export class MailCoreCommon {
                     continue;
                 }
                 this.env.bus.trigger("mail.message/delete", { message, notifId });
-                message.delete();
+                if (payload.from_store) {
+                    message.delete();
+                }
```

Comment:
I don't understand the naming, wasn't the relation between "from_store" and "delete" concepts?

Reply:
(delete_)from_store
so if the message needs to be deleted from Store.
Too obscure, I'll change it to `delete_from_store`